### PR TITLE
Separate x and y flags for AxisItem.setLogMode

### PR DESCRIPTION
**Pull request by max-radin**
This fix allows the example code from #997 to be run. I also checked that the logAxis.py example looks the same after this change.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ PyQtGraph developers are highly encouraged to (but not required) to use [`pre-co
 To run the test suite, after installing the above dependencies run
 
 ```bash
-python -m pytest examples tests
+python -m pytest pyqtgraph/examples tests
```
Comment (j9ac9k) [inline CONTRIBUTING.md:71]: nice catch!

**Comment by NilsNemitz** [comment]
Hi @max-radin,

That is interesting, because I think the code that seems to break here is newer than the issue #997.
Ooops. Might have broken something there.

I am currently looking at the log-mode propagation from another angle... I currently think that it might be better to tell the ViewBox about the requested log mode settings from inside `PlotItem.updateLogMode` that should have the information for both axes. The AxisItems don't really have a need to know about both orientations.

Doesn't each of the four potential axes set both x and y mode individually, now?

**Comment by NilsNemitz** [comment]
Ah, sorry, I missed the actual broken bit here.

While I am not sure if `setLogMode` is ever called directly, do you think this can work so that 
- two arguments are treated as `x` and `y`, and AxisItem.logMode is extracted from the correct one, like you have now
- a single argument directly controls `AxisItem.logMode` as before
?

**Comment by max-radin** [comment]
Hi @NilsNemitz, one way we could support the old single-argument behavior would be to do something like
```python
def setLogMode(self, *args, **kwargs):
  if len(args) == 1 or 'log' in kwargs:
    # do the old thing
  else:
    # do the new thing
```
That way calling `setLogMode(True)` will control log mode along the direction of the axis while still allowing `setLogMode(x=True)` to control log mode in the x direction.

Pros:
* Backwards compatibility with any user code that called `AxisItem.logMode`
* Provides an easy way to set the log mode along an axis without having to check its orientation

Cons:
* The behavior of `AxisItem.logMode` will be different from `PlotItem.logMode`.
* The function signature and logic in `AxisItem.logMode` will be a bit more complicated.

Do you think this makes sense to do? I'm pretty new to the codebase so I don't really have a sense of how these methods typically get used.

**Comment by NilsNemitz** [comment]
I think that is a good idea.
You probably do not need to check for the `log` keyword, that is probably not commonly used in a single-parameter function call.
So the simple distinction of one argument / two arguments might make it work.
Then it has the same functionality as other items when called by PlotItem (with two parameters).
And the old single parameter, possibly user-accessible interface still works to set log mode along the axis alignment.

**Comment by max-radin** [comment]
@NilsNemitz just pushed a commit that adds backwards compatibility for the single argument case. thanks!

**Comment by j9ac9k** [comment]
@max-radin thanks for the PR! This LGTM, merging.  @NilsNemitz thanks for reviewing!

**Pull request merged**